MyProject/src/main/java/com/netcracker/controller/InventoryController.java

```diff
@@ -0,0 +1,265 @@
+                ()-> new ObjectNotFoundException("Unable to find network element with id: " + id ));
+
+        if(networkElement.getInterfaces().contains(inter)){
+            throw new IllegalStateException("Network element with id: " + id + " already has assigned interface");
+        }
+
+        Interface anInterface = interfaceRepository.save(inter);
+
+        NeToInterface neToInterface = new NeToInterface(networkElement, anInterface);
+
+        neToInterRepository.save(neToInterface);
+
+        return anInterface;
+    }
+
+
+    @PostMapping("/network-elements/{id}/hardware-component")
+    @ApiOperation(value = "Add hardware component")
+    public HardwareComponent createHardwareComponent(
+            @ApiParam(value = "network element id", required = true)
+            @PathVariable String id,
+            @ApiParam(value = "hardware component object", required = true)
+            @RequestBody HardwareComponent hardwareCom){
+
+        NetworkElement networkElement = neRepository.findById(id).orElseThrow(
+                ()-> new ObjectNotFoundException("Unable to find network element with id: " + id ));
+
+        if(networkElement.getHardwareComponent() != null){
+            throw new IllegalStateException
+                    ("Network element with id: " + id + " already has assigned hardware component");
+        }
+
+        HardwareComponent hwComponent = hwRepository.save(hardwareCom);
+
+        NeToHw neToHw = new NeToHw(networkElement, hwComponent);
+
+        neToHwRepository.save(neToHw);
+
+        return hwComponent;
+    }
+
+    @PostMapping("/interfaces/{id}/sub-interface")
+    @ApiOperation(value = "Add sub interface")
+    public  SubInterface createSubInterface(
+            @ApiParam(value = "Interface id", required = true)
+            @PathVariable String id,
+            @ApiParam(value = "sub interface object", required = true)
+            @RequestBody SubInterface subInter){
+
+        Interface inter = interfaceRepository.findById(id).orElseThrow(
+                ()->new ObjectNotFoundException("Unable to find interface with id: " + id));
+        if(inter.getSubInterfaces().contains(subInter)){
+            throw  new IllegalStateException
+                    ("Interface with id: " + id + " already has assigned sub interface");
+        }
+        SubInterface subInterface = subRepository.save(subInter);
+
+        IntToSub intToSub = new IntToSub(inter, subInterface);
+        intToSubRepository.save(intToSub);
+
+        return subInterface;
+    }
+
+    @PostMapping("/networkElement")
+    @ApiOperation(value = "Add network element")
+    public NetworkElement createNetworkElement(
+            @ApiParam(value = "network element object", required = true)
+            @RequestBody NetworkElement networkEl){
+
+        return neRepository.save(networkEl);
+    }
+
+    @PostMapping("/links")
+    @ApiOperation(value = "Add link")
+    public Link createLink(
```

Comment:
Please encapsulate the link creation logic in a separate service. Please check https://github.com/dmfe/network-inventory/blob/master/src/main/java/com/example/inventoryapp/services/impl/InventoryCRUDService.java
and use it as an example.